crates/sui-transaction-checks/src/lib.rs

```diff
@@ -418,13 +439,17 @@ mod checked {
     ///
     /// Before this function is invoked, TransactionManager must ensure all depended
     /// objects are present. Thus any missing object will panic.
-    pub fn check_sequenced_input_objects<S: ObjectStore + MarkerTableQuery, G: GetSharedLocks>(
+    pub fn check_sequenced_input_objects<S, G>(
```

Comment:
we're going to have some hairy conflicts between this and https://github.com/MystenLabs/sui/pull/14579